src/MCPClient/lib/clientScripts/extract_bag_transfer.py

```diff
@@ -96,14 +96,29 @@ def call(jobs):
                     job.set_status(exit_code)
                     continue
 
-                # checkForTopLevelBag
-                listdir = os.listdir(destinationDirectory)
+                # Ensure that the only thing in the destination dir is the
+                # top level directory from the extracted file, with the
+                # exception of certain files that may have been copied here
+                # previously. (For now this is just the processing config.)
+                # These files will need to be moved down a level.
+                preexisting_files = {"processingMCP.xml"}
+
+                listdir = set(os.listdir(destinationDirectory))
+                to_move = listdir & preexisting_files
+                listdir -= preexisting_files
+
                 if len(listdir) == 1:
-                    internalBagName = listdir[0]
+                    internalBagName = listdir.pop()
+
+                    for filename in to_move:
+                        shutil.move(
+                            os.path.join(destinationDirectory, filename),
+                            os.path.join(destinationDirectory, internalBagName),
+                        )
```

Comment:
When MCPServer starts the transfer and needs to include a non-default processing config, it creates a directory named `SimpleBagWithProcessingMCP` and places the config inside it. So we have:
```
SimpleBagWithProcessingMCP/
├── processingMCP.xml
```
When the zipfile is extracted, we end up with
```
SimpleBagWithProcessingMCP/
├── processingMCP.xml
│   ├── SimpleBagWithProcessingMCP/
│   ├── bag-info.txt
│   ├── bagit.txt
│   ├── data
│   │   ├── LICENSE
│   │   ├── processingMCP.xml
│   │   ├── README
│   │   └── SumiyoshiHonsha.png
│   ├── manifest-sha256.txt
│   ├── manifest-sha512.txt
│   ├── tagmanifest-sha256.txt
│   └── tagmanifest-sha512.txt
```

But that len==0 check aims to make sure there's only one single directory at the root of the bag, and we have the processing MCP there too. So we move it down a level. This is the same place it will end up with non-zipped transfers.

Does that make sense?

Reply:
I think so? Would I be right in saying this would happen when a user starts a transfer from the API and specifies a processing config there?

Reply:
Yes, that's right :)

Reply:
Perfect! Thanks @helenst